src/magic_folder/invite.py

```diff
@@ -5,181 +5,95 @@
-        raise e
-
-    from_url = nodeurl.child(
-        u"uri",
-        u"{}".format(rootcap),
-    )
-
-    if from_path:
-        from_url = from_url.child(
-            u"{}".format(escape_path(from_path))
-        )
-
-    from_url = from_url.add(
-        u"t",
-        u"json",
-    )
-
-    get_url = from_url.to_uri().to_text().encode("ascii")
-    response = yield treq.get(get_url)
-    if response.code != OK:
-        returnValue((yield bad_response(get_url, response)))
-    result = yield readBody(response)
-
-    nodetype, attrs = json.loads(result)
-    cap = to_str(attrs.get("rw_uri") or attrs["ro_uri"])
-
-    # now get the target
-    try:
-        rootcap, path = get_alias(aliases, to_file, DEFAULT_ALIAS)
-    except Exception as e:
-        raise e
-
-    to_url = nodeurl.child(
-        u"uri",
-        u"{}".format(rootcap)
-    )
-
-    if path:
-        to_url = to_url.child(
-            u"{}".format(escape_path(path))
-        )
-
-    if path.endswith("/"):
-        # "mv foo.txt bar/" == "mv foo.txt bar/foo.txt"
-        to_url = to_url.child(
-            u"{}".format(escape_path(from_path[from_path.rfind("/")+1:]))
-        )
-
-    put_url = to_url.add(
-        u"t",
-        u"uri",
-    ).add(
-        u"replace",
-        u"only-files",
-    )
-
-    response = yield treq.put(put_url.to_text(), cap)
-    if not re.search(r'^2\d\d$', str(response.code)):
-        if response.code == CONFLICT:
-            raise Exception("You cannot overwrite a directory with a file")
-        else:
-            raise Exception(format_http_error("Error", response))
-
-    returnValue(0)
+from .tahoe_client import (
+    create_tahoe_client,
+)
 
 
 @inlineCallbacks
-def magic_folder_invite(node_directory, alias, nickname, treq):
+def magic_folder_invite(config, folder_name, invitee_name, treq):
     """
     Invite a user identified by the nickname to a folder owned by the alias
 
-    :param unicode node_directory: The root of the Tahoe-LAFS node.
+    :param GlobalConfigDatabase config: our configuration
 
-    :param unicode alias: The alias of the folder to which the invitation is
-        being generated.
+    :param unicode folder_name: The name of an existing magic-folder
 
-    :param unicode nickname: The nickname of the invitee.
+    :param unicode invitee_name: The nickname for the new participant
 
     :param HTTPClient treq: An ``HTTPClient`` or similar object to use to make
         the queries.
 
     :return Deferred[unicode]: A secret invitation code.
     """
-    aliases = get_aliases(node_directory)[alias]
-    nodeurl = get_node_url(node_directory)
+    # XXX probably want to pass this in, instead of "treq"?
+    tahoe_client = create_tahoe_client(config.tahoe_client_url, treq)
+
+    # get configuration for this magic-folder (or error if it doesn't
+    # exist)
+    try:
+        folder_config = config.get_magic_folder(folder_name)
+    except ValueError:
+        raise usage.UsageError(
```

Comment:
Yes, good point .. I also just deleted all this because we want magic-wormhole invites though (but yes I guess that means `cli.py` should be the only place `usage` use exists).